openedx/features/genplus_features/genplus_learning/api/v1/views.py

```diff
@@ -69,3 +72,50 @@ def get_queryset(self):
         except Class.DoesNotExist:
             return Student.objects.none()
         return gen_class.students.select_related('gen_user__user').all()
+
+
+class ClassSummaryViewSet(mixins.RetrieveModelMixin,
+                          viewsets.GenericViewSet):
+    authentication_classes = [SessionAuthenticationCrossDomainCsrf]
+    permission_classes = [IsAuthenticated, IsTeacher]
+    serializer_class = ClassSummarySerializer
+    queryset = Class.objects.all()
+    lookup_field = 'group_id'
+
+    def retrieve(self, request, group_id=None):  # pylint: disable=unused-argument
+        """
+        Returns the summary for a Class
+        """
+        gen_class = self.get_object()
+        class_units = ClassUnit.objects.select_related('gen_class', 'unit').prefetch_related('class_lessons')
+        class_units = class_units.filter(gen_class=gen_class)
+        data = self.get_serializer(class_units, many=True).data
+
+        for i in range(len(data)):
+            lessons = data[i]['class_lessons']
+            data[i]['unit_progress'] = round(statistics.fmean([lesson['class_lesson_progress']
+                                                               for lesson in lessons])) if lessons else 0
+
+        gen_class_data = {
+            'school_name': gen_class.school.name,
+            'class_name': gen_class.name,
+            'class_image': get_absolute_url(request, gen_class.image),
+            'results': data,
+        }
+
+        return Response(gen_class_data, status=status.HTTP_200_OK)
+
+    @action(detail=True, methods=['put'])
+    def unlock_lesson(self, request, lesson_id=None):  # pylint: disable=unused-argument
+        """
+       unlock the lesson of the unit
+        """
+        try:
```

Comment:
can we use this 
from django.shortcuts import get_object_or_404
instead of try catch